### PR TITLE
[cli] prebuild: only alter scripts when they match default template values

### LINKED ISSUE #29446: Prebuild command should not add "ios" and "android" scripts to package.json
### Summary

Every time I run `npx expo prebuild`, two new scripts `ios` and `android` scripts gets added to the package.json.

I've purposely removed them because I like to use a different set of scripts. Now every time I run `prebuild` I have to go in and remove the script entries again. I don't think the `prebuild` command should add these scripts. It's fine to have them in a new project template, but developers should be allowed to change them as they see fit.

### What platform(s) does this occur on?

Android, iOS, Web

### SDK Version

51

### Environment

```
 expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.2.0 - ~/.asdf/installs/nodejs/20.2.0/bin/node
      npm: 9.6.6 - ~/.asdf/plugins/nodejs/shims/npm
    Managers:
      CocoaPods: 1.14.3 - /Users/ward/.asdf/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2023.2 AI-232.10300.40.2321.11668458
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.7 => 51.0.7 
      react: 18.2.0 => 18.2.0 
      react-native: 0.74.1 => 0.74.1 
    Expo Workflow: managed
```

### Minimal reproducible example

1.  Create a fresh expo project
2. Remove the `ios` and `android` scripts from package.json
3. Run `npx expo prebuild`

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-12448
Fixes #29446
* #29446

# How

When running `prebuild` only alter scripts when they match default template values.

# Test Plan

Add unit tests, make sure they passes.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by bsmayer** [comment]
From what I understood from this PR, we still wouldn't have the option to omit the `android` and `ios` scripts entirely, right? 
The reason why I'm saying that is because on our team we have `ios:[staging|production]` and `android:[staging|production]` as a pattern and we would like to get rid of the default commands generated by prebuild, but no matter what I do, every time we run a prebuild, it adds the scripts back.

**Comment by Simek** [comment]
The initial version of this changeset was only altering the commands, but that exposed the fact that we are using and testing the append functionality (when scripts are not present) in our tests suits and few E2E tests. 

Let me gather a bit more of context and history on this functionality, so we are sure that the removal won't affect other users in a negative way, before we proceed with stripping down append logic and tests rewrite.

File: packages/@expo/cli/src/prebuild/updatePackageJson.ts [hunk 1]
```diff
@@ -66,9 +66,9 @@ export async function updatePackageJSONAsync(
  * 3. Update `package.json` `main`.
  *
  * @param projectRoot The root directory of the project.
- * @param props.templatePkg Template project package.json as JSON.
- * @param props.pkg Current package.json as JSON.
- * @param props.skipDependencyUpdate Array of dependencies to skip updating.
```
Comment (byCedric) [inline packages/@expo/cli/src/prebuild/updatePackageJson.ts:71]: I think these `props.x` notations were correct JSDoc parameters, why did you remove the `props.` prefix?

image

**Review by byCedric — APPROVED** [review]
Looks good, thanks! Left 1 comment

File: packages/@expo/cli/src/prebuild/updatePackageJson.ts (hunk 1 above)
Comment (Simek) [reply packages/@expo/cli/src/prebuild/updatePackageJson.ts:71]: This notation only applies to named parameters, and if using destructure notation `options` is a correct default name, so TL/DR it could be `options.templatePkg` or `templatePkg`, but not `props.templatePkg` unless we name the parameter manually as `props`.

Also, we are using TypeDoc, which relies on a mix of JSDoc, TSDoc and their own standard, so we should in most cases refer to their docs instead:
* https://typedoc.org/tags/param/

File: packages/@expo/cli/src/prebuild/updatePackageJson.ts [hunk 2]
```diff
@@ -239,19 +239,19 @@ export function createDependenciesMap(dependencies: any): DependenciesMap {
...
+ * the default values used in project templates.
  */
-function updatePkgScripts({ pkg }: { pkg: PackageJSONConfig }) {
+export function updatePkgScripts({ pkg }: { pkg: PackageJSONConfig }) {
   let hasChanged = false;
   if (!pkg.scripts) {
     pkg.scripts = {};
   }
-  if (!pkg.scripts.android?.includes('run')) {
+  if (!pkg.scripts.android || pkg.scripts.android === 'expo start --android') {
```
Comment (EvanBacon) [inline packages/@expo/cli/src/prebuild/updatePackageJson.ts:250]: We used a match against "run" so users could keep community CLI scripts in place. The exact match seems a bit fragile.

**Review by brentvatne — APPROVED** [review]
